Runtime/Scripts/Context.cs

```diff
@@ -54,6 +55,69 @@ internal static void Quit()
         }
     }
 
+    internal class Batch
+    {
+        public struct BatchData
+        {
+            public int tracksCount;
+            public IntPtr[] tracks;
+        }
+
+        public BatchData data;
+        public IntPtr ptr;
+
+        public Batch()
+        {
+            ResizeCapacity(1);
+        }
+
+        ~Batch()
+        {
+            this.Dispose();
+        }
+
+        public void Dispose()
+        {
+            if (ptr != IntPtr.Zero)
+            {
+                Marshal.FreeHGlobal(ptr);
+                ptr = IntPtr.Zero;
+            }
+        }
+
+        public void ResizeCapacity(int totalTracks)
+        {
+            const int roundedCapacity = 32;
+            int totalCapacity = ((totalTracks + roundedCapacity) / roundedCapacity) * roundedCapacity;
+
+            data.tracksCount = 0;
+            data.tracks = new IntPtr[totalCapacity];
+
+            int size = Marshal.SizeOf(typeof(BatchData)) +
+                       Marshal.SizeOf(typeof(IntPtr)) * data.tracks.Length;
+
+            if (ptr == IntPtr.Zero)
+                ptr = Marshal.AllocHGlobal(size);
+            else
+                ptr = Marshal.ReAllocHGlobal(ptr, (IntPtr)size);
+            Marshal.StructureToPtr(data, ptr, false);
+        }
```

Comment:
I think this will resize even if the current size is big enough.

I think we could track the current totalCapacity, and only resize when the requested capacity is bigger.
Doing this could avoid unncessary re-allocations.

Reply:
Besides the constructor, the other place where ResizeCapacity gets called is guarded by a < length check, but I can add something like

```
            if (ptr != IntPtr.Zero && data.tracks.Length == totalCapacity)
                return;
```

and alternatively take the higher level size check away?

Reply:
Modified in https://github.com/Unity-Technologies/com.unity.webrtc/pull/915/commits/10b9e72a561a6142833db9a8cca0a4e1c9aa9511